src/snowflake/snowpark/column.py

```diff
@@ -923,14 +923,19 @@ def name(self, alias: str, variant_is_as: bool = None) -> "Column":
                 ast.variant_is_as.value = variant_is_as
         return Column(Alias(self._expression, quote_name(alias)), ast=expr)
 
-    # TODO: Implement AST generation for Column.over (relies on SpWindowSpec AST generation)
     def over(self, window: Optional[WindowSpec] = None) -> "Column":
         """
         Returns a window frame, based on the specified :class:`~snowflake.snowpark.window.WindowSpec`.
         """
         if not window:
             window = Window._spec()
-        return window._with_aggregate(self._expression)
+
+        ast = proto.Expr()
+        expr = with_src_position(ast.sp_column_over)
+        expr.col.CopyFrom(self._ast)
+        expr.window_spec.CopyFrom(window._ast if window else WindowSpec())
```

Comment:
Now that WindowSpecs are exprs, we no longer need to use CopyFrom. Each WindowSpec can generate its own Assign() and here we can use `SpWindowSpecRef(id)`. Alternatively, if you want to keep this model, then the AST should go back to using a `data` decl describing a sum type.

Reply:
Hi, following up on:

> Now that WindowSpecs are exprs, we no longer need to use CopyFrom. Each WindowSpec can generate its own Assign() and here we can use SpWindowSpecRef(id). Alternatively, if you want to keep this model, then the AST should go back to using a data decl describing a sum type.

this would mean for
```
window1 = Window.partition_by("value").order_by("key").rows_between(Window.CURRENT_ROW, 2)
df.select(avg("value").over(window1))
```
we would encode this as
```
Assign(uid=1, var_id=1, expr=SpWindowSpecRowBetween(wnd=SpWindowSpecOrderBy(wnd=SpWindowSpecPartitionBy(wnd=SpWindowSpec(),...), ...) , ...))

Assign(..., expr=SpDataFrameSelect(..., expr=SpColumnOver(SpWindowSpecRef(id=1)))
```
?

Or do you have in mind here
```
Assign(uid=1000, var_id=1, expr=SpWindowSpec())
Assign(uid=1001, var_id=2, expr=SpWindowSpecPartitionBy(wnd=SpWindowSpecRef(id=1), ...))
...
```
?

Reply:
discussed offline: Solution is to keep it aligned with `Column`, however `SpWindowSpec` should not be an `Expr` but rather inherit the SrcPosition trait. Will make changes accordingly.